test/test-PerfMeter.js

```diff
@@ -0,0 +1,162 @@
+/* globals Services */
+
+"use strict";
+const simplePrefs = require("sdk/simple-prefs");
+
+const {before, after} = require("sdk/test/utils");
+const tabs = require("sdk/tabs");
+const {setTimeout} = require("sdk/timers");
+const {ActivityStreams} = require("lib/ActivityStreams");
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/Services.jsm");
+
+let ACTIVITY_STREAMS_URL;
+let app;
+
+function openTestTab(notifyEvent) {
+  let tabData = {};
+  tabData.showPromise = new Promise(resolve => {
+    function onShow(tab) {
+      tabData.tab = tab;
+      tabs.removeListener("pageshow", onShow);
+      if (!notifyEvent) {
+        setTimeout(function() {
+          resolve();
```

Comment:
Not sure i understand the question, but if you wonder why the promise is resolved inside setTimeout, see the previous comment